api/v1/postgres_types.go

```diff
@@ -575,77 +507,42 @@ func init() {
 	SchemeBuilder.Register(&Postgres{}, &PostgresList{})
 }
 
+func (p *Postgres) buildAdditionalVolumes(c *corev1.ConfigMap) []zalando.AdditionalVolume {
+	if c == nil {
+		// abort if the global configmap is not there
+		return nil
+	}
+
+	// Unmarshal yaml-string of additional volumes
+	volumes := &[]zalando.AdditionalVolume{}
+	if err := yaml.Unmarshal([]byte(c.Data["additional-volumes"]), volumes); err != nil {
+		return nil
+	}
+
+	return *volumes
+}
+
 func (p *Postgres) buildSidecars(c *corev1.ConfigMap) []zalando.Sidecar {
 	if c == nil {
 		// abort if the global configmap is not there
 		return nil
 	}
 
-	exporterContainerPort, error := strconv.ParseInt(c.Data["postgres-exporter-container-port"], 10, 32)
-	if error != nil {
-		// todo log error
-		exporterContainerPort = 9187
+	// Unmarshal yaml-string of exporter
+	sidecars := []zalando.Sidecar{}
+	if err := yaml.Unmarshal([]byte(c.Data["sidecars"]), &sidecars); err != nil {
+		return nil
 	}
-	return []zalando.Sidecar{
-		{
-			Name:        ExporterSidecarName,
-			DockerImage: c.Data["postgres-exporter-image"],
-			Ports: []corev1.ContainerPort{
-				{
-					Name:          ExporterSidecarPortName.StrVal,
-					ContainerPort: int32(exporterContainerPort),
-					Protocol:      corev1.ProtocolTCP,
-				},
-			},
-			Resources: zalando.Resources{
-				ResourceLimits: zalando.ResourceDescription{
-					CPU:    c.Data["postgres-exporter-limits-cpu"],
-					Memory: c.Data["postgres-exporter-limits-memory"],
-				},
-				ResourceRequests: zalando.ResourceDescription{
-					CPU:    c.Data["postgres-exporter-requests-cpu"],
-					Memory: c.Data["postgres-exporter-requests-memory"],
-				},
-			},
-			Env: []corev1.EnvVar{
-				{
-					Name:  "DATA_SOURCE_URI",
-					Value: "127.0.0.1:5432/postgres?sslmode=disable",
-				},
-				{
-					Name:  "DATA_SOURCE_USER",
-					Value: "postgres",
-				},
-				{
-					Name: "DATA_SOURCE_PASS",
-					ValueFrom: &corev1.EnvVarSource{
-						SecretKeyRef: &corev1.SecretKeySelector{
-							LocalObjectReference: corev1.LocalObjectReference{
-								Name: "postgres." + p.ToPeripheralResourceName() + ".credentials",
-							},
-							Key: "password",
-						},
-					},
-				},
-				{
-					Name:  "PG_EXPORTER_EXTEND_QUERY_PATH",
-					Value: "/metrics/queries.yaml",
-				},
-			},
-		},
-		{
-			Name:        FluentBitSidecarName,
-			DockerImage: c.Data["postgres-fluentbit-image"],
-			Resources: zalando.Resources{
-				ResourceLimits: zalando.ResourceDescription{
-					CPU:    c.Data["postgres-fluentbit-limits-cpu"],
-					Memory: c.Data["postgres-fluentbit-limits-memory"],
-				},
-				ResourceRequests: zalando.ResourceDescription{
-					CPU:    c.Data["postgres-fluentbit-requests-cpu"],
-					Memory: c.Data["postgres-fluentbit-requests-memory"],
-				},
-			},
-		},
+
+	// Deal with dynamically assigned name
+	for i := range sidecars {
+		for j := range sidecars[i].Env {
+			if sidecars[i].Env[j].Name == "DATA_SOURCE_PASS" {
```

Comment:
this check should be more general: If there is a sidecars[i].Env[j].ValueFrom, we should then check if it is of the kind `SecretKeyRef`, and if so, we should proceed.

This implementation is just a workaround for one specific case.